gateway/middleware.go

```diff
@@ -4,6 +4,15 @@ import (
 	"net/http"
 
 	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var requestCounter = prometheus.NewCounter(
```

Comment:
```suggestion
var requestDuration = prometheus.NewHistogramVec(
```
make this a histogram